src/app/add-clients/helperFunctions.tsx

```diff
@@ -0,0 +1,254 @@
+import React from "react";
+import { Database } from "@/database_types_file";
+import { booleanGroup } from "@/components/DataInput/inputHandlerFactories";
+import { SelectChangeEvent } from "@mui/material";
+import { InsertSchema } from "@/supabase";
+import supabase from "@/supabase";
+
+type PersonType = Database["public"]["Enums"]["gender"];
+type FieldType = string | (boolean | null) | booleanGroup | Address | People[];
+type EventType = React.ChangeEvent<HTMLInputElement> | SelectChangeEvent;
+type OnChangeType = (event: EventType) => void;
+type SetErrorType = (errorKey: string, errorMessage: string) => void;
+type SetFieldType = (fieldKey: string, newFieldValue: FieldType) => void;
+export type ClientDatabaseRecord = InsertSchema["clients"];
+type FamilyDatabaseRecord = InsertSchema["families"];
+
+interface Address {
+    line1: string;
+    line2: string;
+    town: string;
+    county: string;
+    postcode: string;
+}
+
+export interface People {
+    personType: PersonType;
```

Comment:
gender instead?

Reply:
Same here, just to match what's on the database :)